src/Twig/GroupRuntime.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Twig;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Twig\Extension\RuntimeExtensionInterface;
+
+/**
+ * @phpstan-import-type Item from \Sonata\AdminBundle\Admin\Pool
+ */
+final class GroupRuntime implements RuntimeExtensionInterface
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @internal This class should only be used through Twig
+     */
+    public function __construct(Pool $pool)
+    {
+        $this->pool = $pool;
+    }
+
+    /**
+     * @phpstan-return array<array{
+     *  label: string,
+     *  label_catalogue: string,
+     *  icon: string,
+     *  item_adds: Item[],
+     *  items: array<AdminInterface<object>>,
+     *  keep_open: bool,
+     *  on_top: bool,
+     *  roles: list<string>
+     * }>
+     */
+    public function getDashboardGroupsWithCreatableAdmins(): array
+    {
+        $groups = [];
+
+        foreach ($this->pool->getDashboardGroups() as $group) {
+            $filteredGroups = array_filter($group['items'], static function (AdminInterface $admin): bool {
```

Comment:
As we are only looking the array for a single occurrence (we can stop the search at the first match), I think using `array_filter()` is expensive as it will search in the whole array.

Reply:
Changed